### PR TITLE
Improve docs about how to write guides

### LINKED ISSUE #420: More docs for guide authors
We could use more docs describing the various features used in the guides. README.md is a good start but once I started writing guides I learned a lot by reading code and from PR feedback, and a lot of that should be written up briefly to help new authors get up to speed.

Some general areas to consider:
- document all macros
- a brief description of generally how to add a new macro  
- document all replacements (e.g. `@fooVersion@`)
- describe the build process, how the asciidoc is pre-processed to the final asciidoc, etc.
- Sergio's new shared src folder feature (https://github.com/micronaut-projects/micronaut-guides/pull/404)
- some notes on dependencies, e.g. setting the version in pom.xml properties
- the new Rocker template support that I added
- guide-specific notes on brand guidelines
- metadata.json options

### PR DESCRIPTION
**Pull request by ilopmar**
Fixes #420

### CODE REVIEW

**Comment by ilopmar** [comment]
I've added both @burtbeckwith and @sdelamo for reviewers because both of you have written a lot of guides. I've also added @alvarosanchez because I explained yesterday how this works, so he can check this docs and see if they make sense.

**Review by burtbeckwith — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,29 +1,280 @@
...
+For a Gradle guide, the first block will be included. For a Maven guide, the second one will be included.
+
+As before, look for usages of the macro in the `guides` directory for more examples.
+
+### New Guide Template
+
+To create a new guide use the following template as the base asciidoc file:
 
 ```asciidoc
 include::{commondir}/common-header.adoc[]
```
Comment (sdelamo) [inline README.md]: This needs to be updated from

`include::{commondir}/common-header.adoc[]`

to: `common:header.adoc[]`

see: https://github.com/micronaut-projects/micronaut-guides/pull/688

File: README.md [hunk 2]
```diff
@@ -60,7 +311,7 @@ TODO Use the generic next step
 
 include::{commondir}/common-next.adoc[]
```
Comment (sdelamo) [inline README.md]: This needs to be updated from

`include::{commondir}/common-next.adoc[]`

to: `common:next.adoc[]`

see: https://github.com/micronaut-projects/micronaut-guides/pull/688

File: README.md [hunk 3]
```diff
@@ -69,9 +320,27 @@ TODO: link to the documentation modules you used in the guide
 include::{commondir}/common-helpWithMicronaut.adoc[]
```
Comment (sdelamo) [inline README.md]: remove 

`include::{commondir}/common-helpWithMicronaut.adoc[]` to avoid Oracle members including "OCI sponsored this guide" callout.

**Review by sdelamo — CHANGES_REQUESTED** [review]
It looks great. Many thanks for writing this.

**Pull request merged**